workflow/artifacts/hdfs/util.go

```diff
@@ -8,9 +8,10 @@ import (
 	"github.com/jcmturner/gokrb5/v8/config"
 )
 
-func createHDFSClient(addresses []string, user string, krbOptions *KrbOptions) (*hdfs.Client, error) {
+func createHDFSClient(addresses []string, user string, dataTransferProtection string, krbOptions *KrbOptions) (*hdfs.Client, error) {
 	options := hdfs.ClientOptions{
-		Addresses: addresses,
+		Addresses:              addresses,
+		DataTransferProtection: dataTransferProtection,
```

Comment:
To make artifacts more sustainable (other than plugins #5862), I've been arguing we should perhaps accept arbitrary YAML string output. E.g. in this case:
```yaml
clientOptions: |
  dataTransferProtection: true
```

That way we don't have to add a new feature every time a user wants to configure something else